code/modules/projectiles/gun_system.dm

```diff
@@ -471,17 +500,89 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 			//							    \\
 			//						   	    \\
 //----------------------------------------------------------
+///Check if the gun can fire and add it to bucket auto_fire system if needed, or just fire the gun if not
+/obj/item/weapon/gun/proc/start_fire(datum/source, atom/object, turf/location, control, params)
+	SIGNAL_HANDLER
+	if(gun_on_cooldown(gun_user))
+		return
+	if(gun_user.hand && !isgun(gun_user.l_hand) || !gun_user.hand && !isgun(gun_user.r_hand)) // If the object in our active hand is not a gun, abort
+		return
+	if(gun_user.hand && isgun(gun_user.r_hand) || !gun_user.hand && isgun(gun_user.l_hand)) // If we have a gun in our inactive hand too, both guns get innacuracy maluses
+		dual_wield = TRUE
```

Comment:
This is actually much better IMO.
The old system applied the malus only to the gun that was in the inactive hand, now it is applied on both gun.
This system also takes into account the fire rate / fire mode of each gun, while in old system you could just take a gun with a low ROF, and used it in akimbo with a slow ROF gun. The slow ROF gun could then be shot with the ROF of the first gun in autofire, even if the slow ROF gun did not have autofire.

That being said, it's also a balance change, and we should probably reduce the maluses from makimbo to compensate

Reply:
akimbo stuff is wonkers yes and its balance is terrible, discuss later on discord because im awoozy right now